src/OpenTelemetry.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -135,6 +140,16 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         }
     }
 
+    if (use_dotnet_startuphook_bootstrapper)
+    {
+        const auto home_path = GetEnvironmentValue(environment::profiler_home_path);
+        const auto startup_hooks = GetEnvironmentValue(environment::dotnet_startup_hooks);
+        if (!IsStartupHookEnabled(startup_hooks, home_path))
+        {
+          Logger::Error("The required startup hook was not configured correctly. No telemetry will be captured.");
```

Comment:
Since "no telemetry will be capture" shouldn't it return E_FAIL here?

Reply:
That's a good idea. I was initially thinking that the byte code instrumentation would still do something, but there won't be anything to consume that data.